osutil/user.go

```diff
@@ -55,6 +55,123 @@ type AddUserOptions struct {
 // allows as valid usernames
 var IsValidUsername = regexp.MustCompile(`^[a-z0-9][-a-z0-9+._]*$`).MatchString
 
+// EnsureUserGroup uses the standard shadow utilities' 'useradd' and 'groupadd'
+// commands for creating non-login system users and groups that is portable
+// cross-distro. It will create the group with groupname 'name' and gid 'id' as
+// well as the user with username 'name' and uid 'id'. Importantly, 'useradd'
+// and 'groupadd' will use NSS to determine if a uid/gid is already assigned
+// (so LDAP, etc are consulted), but will themselves only add to local files,
+// which is exactly what we want since we don't want snaps to be blocked on
+// LDAP, etc when performing lookups.
+func EnsureUserGroup(name string, id uint32, extraUsers bool) error {
+	if !IsValidUsername(name) {
+		return fmt.Errorf(`cannot add user/group %q: name contains invalid characters`, name)
+	}
+
+	// Perform uid and gid lookups (with extrausers support)
+	uid, uidErr := FindUidGetent(name)
+	if uidErr != nil {
```

Comment:
(nitpick^2) Maybe a small helper like: IsUnknown{User,Group}(err error)` and then we could write:
```
	if uidErr != nil && !IsUnknownUser(uidErr) {
              ....
```
something like: http://paste.ubuntu.com/p/RgjHgPxr5Y/

Reply:
Done